integration-tests/src/tests.rs

```diff
@@ -515,3 +516,54 @@ fn requires_future_cargo_web_cfg_not_dep() {
     assert_fails_to_build( Wasm32UnknownUnknown, "req-future-cargo-web-cfg-not-dep" );
     assert_builds( Wasm32UnknownEmscripten, "req-future-cargo-web-cfg-not-dep" );
 }
+
+#[test]
+fn mount_path_fails_for_top_level() {
+    // only applies to wasm32-unknown-unknown backend
+    let cwd = crate_path( "mount-path" );
+    assert_fails_to_build( Wasm32UnknownEmscripten, "mount-path" );
+    assert_fails_to_build( AsmjsUnknownEmscripten, "mount-path" );
+    assert_fails_to_build( Wasm32UnknownUnknown, "mount-path" );
+}
+
+
+#[test]
+fn mount_path_for_wasm32_unknown_unknown() {
+    // only applies to wasm32-unknown-unknown backend
+    let cwd = crate_path( "mount-path-wasm32-unknown-unknown" );
+    assert_builds( Wasm32UnknownUnknown, "mount-path-wasm32-unknown-unknown" );
+    let build_dir = "release";
+    let output = cwd.join( "target" ).join( Wasm32UnknownUnknown.to_str() ).join( build_dir ).join( "mount-path.js" );
+    assert_file_contains( output, "/custom/static/" );
+}
+
+#[cfg_attr(not(test_wasm32_unknown_unknown), ignore)]
+#[test]
+fn mount_path_static_files_serve() {
+    let cwd = crate_path( "mount-path-wasm32-unknown-unknown" );
+    use reqwest::header::CONTENT_TYPE;
+    use reqwest::StatusCode;
+
+    run( &cwd, &*CARGO_WEB, &["build", "--target", "wasm32-unknown-unknown"] ).assert_success();
+    let _child = run_in_the_background( &cwd, &*CARGO_WEB, &["start", "--target", "wasm32-unknown-unknown"] );
+
+    let start = Instant::now();
+    let mut response = None;
+    while start.elapsed() < Duration::from_secs( 10 ) && response.is_none() {
+        thread::sleep( Duration::from_millis( 100 ) );
+        response = reqwest::get( "http://localhost:8000" ).ok();
+    }
+
+    let mut response = reqwest::get( "http://localhost:8000/mount-path.js" ).unwrap();
+    assert!(response.text().unwrap().contains("/custom/static/"));
+}
+
+#[cfg_attr(not(test_wasm32_unknown_unknown), ignore)]
+#[test]
+fn mount_path_target_wasm32_unknown_unknown_deploy() {
+    let cwd = crate_path( "mount-path-wasm32-unknown-unknown" );
+    run( &cwd, &*CARGO_WEB, &["deploy"] ).assert_success();
+    let build_dir = "deploy";
+    let output = cwd.join( "target" ).join( Wasm32UnknownUnknown.to_str() ).join( build_dir ).join( "mount-path.js" );
```

Comment:
Here this should be deployed to `join( build_dir ).join( "custom" ).join( "static" ).join( "mount-path.js" )`.

The reason for that is that the stuff which `cargo web start` serves is supposed to be generated in an *exactly* the same way by `cargo web deploy`. So if `start` serves up the file at `/custom/static/mount-path.js` then it should also be deployed at `/custom/static/mount-path.js`. The idea is that you should be able to just run a static file web server in the deployed directory and it should behave just as if it would be served by `start`.

Do you actually need the `deploy` to generate the artifacts in a flat way? (In `target/deploy` instead of in `target/deploy/custom/static/` in this case.)